src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -82,7 +99,8 @@ private static void createStatus(@NonNull Run<?, ?> build, @NonNull TaskListener
         try {
             url = getRootURL(build);
         } catch (IllegalStateException e) {
-            listener.getLogger().println("Can not determine Jenkins root URL. " +
+            listener.getLogger().println("Can not determine Jenkins root URL " +
+                    "or Jenkins URL is not a valid URL regarding Bitbucket API. " +
```

Comment:
Regarding [JENKINS-54297: Failing to report build status to BitBucket server](https://issues.jenkins-ci.org/browse/JENKINS-54297), if you don't log the `IllegalStateException`'s message, we have no way of knowing we need to fix the Jenkins URL to be a FQDN after upgrading to the latest version.

Reply:
it's definitely not a bug, it is a feature that is trying to prevent bad Jenkins URL 😅
The error message is pretty clear, but we welcome contributions 🙏